lib/consensus/isaac.go

```diff
@@ -88,30 +88,42 @@ func (is *ISAAC) SelectProposer(blockHeight uint64, round uint64) string {
 	return is.proposerSelector.Select(blockHeight, round)
 }
 
-func (is *ISAAC) IsValidVotingBasis(basis voting.Basis, latestBlock block.Block) bool {
+//
+// Check if `basis` is a valid one for the current round of consensus
+//
+// Params:
+//   basis = `voting.Basis` received for the current round of consensus
+//   latestBlock = the latest block known to the node
+//
+// Returns:
+//   A string explaining why it's not the latest voting basis,
+//   or empty string ("") if the voting basis is valid.
+//
+func (is *ISAAC) IsValidVotingBasis(basis voting.Basis, latestBlock block.Block) string {
```

Comment:
I think `bool` return is better to understand this method. Because it starts with `Is`.
So, how about returning (bool, string)? like
```
return false, "basis' height is different from latest block's height"
return true, ""
return false, "basis' block hash does not match latest block's hash"
return false, return "basis' round is <= to latest voting basis' round"
return true, ""
```

Reply:
Not a fan of redundant return value (because returning `true` and a non-empty string should never happen), but I can compromise on this :)